File: 1.Two_Sum.py
class Solution(object):
    def twoSum(self, nums:list[int], target:int) -> list:
        checked = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in checked:
                return [checked[diff],i]
            else:
                checked[nums[i]] = i

if __name__ == '__main__':
    l = Solution()
    print(l.twoSum([2,7,11,15], 9))
    print(l.twoSum([3,2,4], 6))
    print(l.twoSum([-1,-2,-3,-4,-5], -8))

File: C++/1.Two_Sum.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> checked;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            int diff = target - nums[i];
            if (checked.find(diff) != checked.end()) {
                return {checked[diff], i};
            } else {
                checked[nums[i]] = i;
            }
        }
        return {};
    }
};

